### PR TITLE
Gulp amdclean

### PR DESCRIPTION
**Pull request by Ebtoulson**
Created a gulp task to un-AMD jquery-autotagging (except for jquery and jquery cookie).
- [ ] js tests pass

### CODE REVIEW

File: Gulpfile.coffee [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    .pipe concat('jquery.autotagging.js')
+    .pipe amdclean.gulp(
+      prefixMode: 'standard'
+      globalModules: ["WH"]
+      prefixTransform: (postName, preName)->
+        switch preName
+          when "jquery" then "jQuery"
+          else postName
+    )
+    .pipe gulp.dest('./dist/non-amd/')
```
Comment (Ebtoulson) [inline Gulpfile.coffee:26]: wasnt sure what to call it so I'm down for suggestions

File: Gulpfile.coffee [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    .src coffee_files
+    .pipe coffee(bare: true)
+    .pipe concat('jquery.autotagging.js')
+    .pipe amdclean.gulp(
+      prefixMode: 'standard'
+      globalModules: ["WH"]
+      prefixTransform: (postName, preName)->
+        switch preName
+          when "jquery" then "jQuery"
+          else postName
```
Comment (Ebtoulson) [inline Gulpfile.coffee:24]: ended up changing the module name after transform to grab the global jQuery. Does anybody know of another approach/solution?

File: src/jquery.autotagging.coffee [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
   'jquery'
   'browserdetect'
-  'underscore'
   './click_handler'
   './select_change_handler'
   'jquery.cookie'
-], ($, browserdetect, _, ClickEventHandler, SelectChangeHandler) ->
-  class WH
+], ($, browserdetect, ClickEventHandler, SelectChangeHandler) ->
+  new class
```
Comment (c0) [inline src/jquery.autotagging.coffee:8]: Note: this changes the API. This class is now a singleton.

**Comment by mpelzsherman** [comment]
Is there a story for this?

Please make sure to bump the major revision number, as this is not going to be backward-compatible (right?).

**Comment by databyte** [comment]
@mpelzsherman - the story ID is in the commit messages. It'll be story #83212064

**Comment by Ebtoulson** [comment]
@mpelzsherman it was for a bug/chore. Originally Portal had to make its own non-AMD version of autotagging so we missed out of important bug fixes and updates so I figured why not create a task that does this for us and then changes would only have to happen once.

I can go ahead and bump the version if you would like.
I've still got some testing to do, but I went again and opened up the PR so I could get everyones option

**Comment by mpelzsherman** [comment]
Sounds like a cool idea! Initially I thought we were converting to non-AMD permanently, not just automating the conversion. So the major version doesn't need to be bumped after all.

**Comment by c0** [comment]
The minor version should still be bumped since it returns a singleton rather than a class. It doesn't have to be in the PR though.

**Comment by Ebtoulson** [comment]
Just an update if anybody wants to look at this while on I'm on ETO. The build tool appears to work but the tests are failing after giving the `jquery.autotagging.coffee` file a `WH` module name.

**Pull request closed without merging**